src/Neo.VM/JumpTable/JumpTable.Compound.cs

```diff
@@ -536,6 +565,14 @@ public virtual void Remove(ExecutionEngine engine, Instruction instruction)
         public virtual void ClearItems(ExecutionEngine engine, Instruction instruction)
         {
             var x = engine.Pop<CompoundType>();
+            if (engine.ReferenceCounter.Version == RCVersion.V2)
+            {
+                foreach (var xSubItem in x.SubItems)
+                {
+                    engine.ReferenceCounter.RemoveStackReference(xSubItem);
+                }
+            }
+
             x.Clear();
```

Comment:
`Clear` may throw an exception in case if `x` is read-only. So technically, there's a diff in the implementation: NeoGo's handler [firstly checks for read-only](https://github.com/nspcc-dev/neo-go/blob/8952922c4d4f4474e390cf3afa4abcdf20c7960d/pkg/vm/vm.go#L1422), then removes refs and then clears `x`. C# implementation firstly removes refs, and then checks for read-only. This difference lead to the fact that VM may end up in FAULT state with different RC value.

From one point, mismatch RC value in FAULT VM is not critical for us, VM is FAULT anyway. From another point, compatibility is always good.

@roman-khimov @shargon @Jim8y what do you think, is it a critical issue?

Reply:
not critical as you said, i am good with either,

Reply:
OK then, let's keep it as is, the code is more simple this way.